validate/validate.go

```diff
@@ -294,10 +294,19 @@ func (v *Validator) CheckMounts() (msgs []string) {
 		return
 	}
 
-	if supportedTypes != nil {
-		for _, mount := range v.spec.Mounts {
-			if !supportedTypes[mount.Type] {
-				msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", mount.Type))
+	for i, mountA := range v.spec.Mounts {
+		if supportedTypes != nil && !supportedTypes[mountA.Type] {
+			msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", mountA.Type))
+		}
+		for j, mountB := range v.spec.Mounts {
+			if nestedValid(v.spec.Platform.OS, mountA.Destination, mountB.Destination) {
+				if v.spec.Platform.OS == "windows" && i < j {
+					msgs = append(msgs, fmt.Sprintf("On Windows, %v nested within %v is forbidden", mountB.Destination, mountA.Destination))
+				}
+				if i > j {
+					// FIXME: add erroring for OS which not support covering mounts
+					logrus.Warnf("On %v, %v will be covered by %v", v.spec.Platform.OS, mountB.Destination, mountA.Destination)
```

Comment:
I'd drop “On %v,”.  Covering is surprising behavior worth warning about regardless of the target OS.  The nesting error a few lines up, on the other hand, is a Windows-specific error, so opening with “On Windows,” is fine.